include/oneapi/dpl/pstl/utils.h

```diff
@@ -279,13 +266,29 @@ class __transform_functor
     _Pred _M_pred;
 
   public:
-    explicit __transform_functor(_Pred __pred) : _M_pred(__pred) {}
+    explicit __transform_functor(_Pred __pred) : _M_pred(std::move(__pred)) {}
 
     template <typename _Input1Type, typename _Input2Type, typename _OutputType>
     void
-    operator()(const _Input1Type& x, const _Input2Type& y, _OutputType& z) const
+    operator()(const _Input1Type& x, const _Input2Type& y, _OutputType&& output) const
```

Comment:
That makes sense, thanks.

Reply:
Apparently, you are right. I changed it to forwarding reference because unary transform functor previously had this behavior. It's again inconsistency with binary transform that I don't understand. But what I read in the standard makes me think that unary transform behavior is correct.